### PR TITLE
DRILL-5896: Handle HBase columns vector creation in the HBaseRecordReader

### PR DESCRIPTION
**Pull request by prasadns14**
Handling vector creation for projected hbase columns, by keeping track of column families and columns separately. 
@paul-rogers please review

### CODE REVIEW

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java [hunk 1]
```diff
@@ -75,6 +75,8 @@
 
   private TableName hbaseTableName;
   private Scan hbaseScan;
+  private Scan hbaseScan1;
+  Set<String> completeFamilies;
```
Comment (paul-rogers) [inline contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: `private`?

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java [hunk 2]
```diff
@@ -87,6 +89,7 @@ public HBaseRecordReader(Connection connection, HBaseSubScan.HBaseSubScanSpec su
     hbaseTableName = TableName.valueOf(
         Preconditions.checkNotNull(subScanSpec, "HBase reader needs a sub-scan spec").getTableName());
     hbaseScan = new Scan(subScanSpec.getStartRow(), subScanSpec.getStopRow());
+    hbaseScan1 = new Scan();
```
Comment (paul-rogers) [inline contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: Better name or comment to explain.

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java [hunk 3]
```diff
@@ -121,16 +125,18 @@ public HBaseRecordReader(Connection connection, HBaseSubScan.HBaseSubScanSpec su
         byte[] family = root.getPath().getBytes();
         transformed.add(SchemaPath.getSimplePath(root.getPath()));
         PathSegment child = root.getChild();
-        if (!completeFamilies.contains(new String(family, StandardCharsets.UTF_8).toLowerCase())) {
-          if (child != null && child.isNamed()) {
-            byte[] qualifier = child.getNameSegment().getPath().getBytes();
+        if (child != null && child.isNamed()) {
+          byte[] qualifier = child.getNameSegment().getPath().getBytes();
+          hbaseScan1.addColumn(family, qualifier);
+          if (!completeFamilies.contains(new String(family, StandardCharsets.UTF_8))) {
```
Comment (paul-rogers) [inline contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: Redundant conversion of `family` to `String`, here and below.

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java [hunk 4]
```diff
@@ -186,6 +192,10 @@ public void setup(OperatorContext context, OutputMutator output) throws Executio
           }
         }
       }
+
+      for (String familyName : completeFamilies) {
+        getOrCreateFamilyVector(familyName, false);
+      }
```
Comment (paul-rogers) [inline contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: Does this create just the map, or also the vectors within the map? Maybe a comment to explain the goals?

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java (hunk 1 above)
Comment (prasadns14) [reply contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: Fixed

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java (hunk 2 above)
Comment (prasadns14) [reply contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: Fixed

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java (hunk 3 above)
Comment (prasadns14) [reply contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: Fixed

File: contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java (hunk 4 above)
Comment (prasadns14) [reply contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java]: It creates only the map vector

**Comment by prasadns14** [comment]
@paul-rogers please review the changes

**Pull request closed without merging**